### PR TITLE
Set telemetry configuration in policy server deployment.

### PR DESCRIPTION
**Pull request by jvanz**
## Description

Updates the telemetry configuration in all the policy servers by default if the Kubewarden controller has telemetry enabled.

Related to https://github.com/kubewarden/ui/issues/287

### CODE REVIEW

File: internal/pkg/admission/policy-server-deployment_test.go [hunk 1]
```diff
@@ -432,3 +432,118 @@ func TestDefaultContainerSecurityContext(t *testing.T) {
...
+		},
+	}
+	deployment := reconciler.deployment("v1", policyServer)
+	hasMetricEnvvar := false
+	hasLogFmtEnvvar := false
+	for _, envvar := range deployment.Spec.Template.Spec.Containers[0].Env {
+		if envvar.Name == "KUBEWARDEN_ENABLE_METRICS" {
+			hasMetricEnvvar = true
+			if envvar.Value != "1" {
+				t.Error("Missing KUBEWARDEN_ENABLE_METRICS value")
```
Comment (viccuad) [inline internal/pkg/admission/policy-server-deployment_test.go:489]: If the variable is set regardless of value, it is enabled. I would reword to:

```suggestion
				t.Error("Present but not reconciled KUBEWARDEN_ENABLE_METRICS value")
```

(same for LOG_FMT)

File: internal/pkg/admission/policy-server-deployment.go [hunk 2]
```diff
@@ -226,6 +226,19 @@ func (r *Reconciler) adaptDeploymentSettingsForPolicyServer(policyServerDeployme
 			},
 		)
 	}
+	if r.MetricsEnabled {
+		policyServerDeployment.Annotations[constants.OptelInjectAnnotation] = "true"
+	}
+}
+
+func envVarsContaintsVariable(container *corev1.Container, envVarName string) (bool, int) {
```
Comment (viccuad) [inline internal/pkg/admission/policy-server-deployment.go:234]:
```suggestion
func envVarsContainVariable(container *corev1.Container, envVarName string) (bool, int) {
```

File: internal/pkg/admission/policy-server-deployment_test.go [hunk 3]
```diff
@@ -432,3 +432,118 @@ func TestDefaultContainerSecurityContext(t *testing.T) {
...
+	for _, envvar := range deployment.Spec.Template.Spec.Containers[0].Env {
+		if envvar.Name == "KUBEWARDEN_ENABLE_METRICS" {
+			hasMetricEnvvar = true
+		}
+		if envvar.Name == "KUBEWARDEN_LOG_FMT" {
+			hasLogFmtEnvvar = true
+		}
+	}
+	if hasMetricEnvvar {
+		t.Error("KUBEWARDEN_ENABLE_METRICS should not define environment variable")
```
Comment (viccuad) [inline internal/pkg/admission/policy-server-deployment_test.go:539]:
```suggestion
		t.Error("KUBEWARDEN_ENABLE_METRICS should not be set")
```

(same for the other messages)

**Review by viccuad — APPROVED** [review]
LGTM! just some nits

File: internal/pkg/admission/policy-server-deployment.go [hunk 4]
```diff
@@ -226,6 +226,19 @@ func (r *Reconciler) adaptDeploymentSettingsForPolicyServer(policyServerDeployme
...
+}
+
+func envVarsContaintVariable(container *corev1.Container, envVarName string) (bool, int) {
+	for i, envvar := range container.Env {
+		if envvar.Name == envVarName {
+			return true, i
+		}
+	}
+	return false, -1
+
```
Comment (flavio) [inline internal/pkg/admission/policy-server-deployment.go:241]: nitpicking...

* The name of the function has a typo. I think you meant to call it `envVarsContainsVariable`
* I don't like the return type of the function, I see two options:
  *  return `int`: it would be `-1` if the item has not been found, otherwise it would return the index position
  * return `(bool, string)`: instead of returning the index to the value, just return the value

I also wonder if it would make sense to make the function more generic by changing the signature to use that as input: `(values []EnvVar, key string)`

File: internal/pkg/admission/policy-server-deployment.go [hunk 5]
```diff
@@ -345,6 +358,23 @@ func (r *Reconciler) deployment(configMapVersion string, policyServer *policiesv
...
 		templateAnnotations = make(map[string]string)
 	}
+	if r.MetricsEnabled {
+		envvar := corev1.EnvVar{Name: constants.PolicyServerEnableMetricsEnvVar, Value: "1"}
+		logFmtEnvVar := corev1.EnvVar{Name: constants.PolicyServerLogFmtEnvVar, Value: "otlp"}
+
+		if hasEvnvar, index := envVarsContaintVariable(&admissionContainer, constants.PolicyServerEnableMetricsEnvVar); hasEvnvar {
+			admissionContainer.Env[index] = envvar
+		} else {
+			admissionContainer.Env = append(admissionContainer.Env, envvar)
```
Comment (flavio) [inline internal/pkg/admission/policy-server-deployment.go:368]: I would move the definition of `envvar` inside of this block

File: internal/pkg/admission/policy-server-deployment.go [hunk 6]
```diff
@@ -345,6 +358,23 @@ func (r *Reconciler) deployment(configMapVersion string, policyServer *policiesv
...
+		}
+
+		if hasEvnvar, index := envVarsContaintVariable(&admissionContainer, constants.PolicyServerLogFmtEnvVar); hasEvnvar {
+			admissionContainer.Env[index] = logFmtEnvVar
+		} else {
+			admissionContainer.Env = append(admissionContainer.Env, logFmtEnvVar)
```
Comment (flavio) [inline internal/pkg/admission/policy-server-deployment.go:374]: I would move the definition of `logFmtEnvVar` into this block

File: internal/pkg/admission/policy-server-deployment_test.go [hunk 7]
```diff
@@ -432,3 +432,118 @@ func TestDefaultContainerSecurityContext(t *testing.T) {
...
+		Spec: policiesv1.PolicyServerSpec{
+			Image: "image",
+			Env:   []corev1.EnvVar{{Name: "KUBEWARDEN_ENABLE_METRICS", Value: "0"}, {Name: "KUBEWARDEN_LOG_FMT", Value: "invalid"}},
+		},
+	}
+	deployment := reconciler.deployment("v1", policyServer)
+	hasMetricEnvvar := false
+	hasLogFmtEnvvar := false
+	for _, envvar := range deployment.Spec.Template.Spec.Containers[0].Env {
+		if envvar.Name == "KUBEWARDEN_ENABLE_METRICS" {
```
Comment (flavio) [inline internal/pkg/admission/policy-server-deployment_test.go:486]: nitpicking, I think we have a constant for that

File: internal/pkg/admission/policy-server-deployment_test.go [hunk 8]
```diff
@@ -432,3 +432,118 @@ func TestDefaultContainerSecurityContext(t *testing.T) {
...
+			hasMetricEnvvar = true
+			if envvar.Value != "1" {
+				t.Error("Present but not reconciled KUBEWARDEN_ENABLE_METRICS value")
+			}
+		}
+		if envvar.Name == "KUBEWARDEN_LOG_FMT" {
```
Comment (flavio) [inline internal/pkg/admission/policy-server-deployment_test.go:492]: nitpicking, I think we have a constant for that

File: internal/pkg/admission/policy-server-deployment_test.go [hunk 9]
```diff
@@ -432,3 +432,118 @@ func TestDefaultContainerSecurityContext(t *testing.T) {
...
+	if !hasMetricEnvvar {
+		t.Error("Missing KUBEWARDEN_ENABLE_METRICS environment variable")
+	}
+	if !hasLogFmtEnvvar {
+		t.Error("Missing KUBEWARDEN_LOG_FMT environment variable")
+	}
+
+	value, hasAnnotation := deployment.Annotations["sidecar.opentelemetry.io/inject"]
+	if !hasAnnotation {
+		t.Error("Missing OPTEL annotation")
```
Comment (flavio) [inline internal/pkg/admission/policy-server-deployment_test.go:508]: I think OPTEL is a typo, I think it should be OTEL

File: internal/pkg/admission/policy-server-deployment_test.go [hunk 10]
```diff
@@ -432,3 +432,118 @@ func TestDefaultContainerSecurityContext(t *testing.T) {
...
+	}
+	if value != "true" {
+		t.Error("OPTEL annotation invalid value")
```
Comment (flavio) [inline internal/pkg/admission/policy-server-deployment_test.go:511]: I think OPTEL is a typo

**Review by flavio — CHANGES_REQUESTED** [review]

File: internal/pkg/admission/policy-server-deployment.go (hunk 5 above)
Comment (jvanz) [reply internal/pkg/admission/policy-server-deployment.go:366]: What do you mean? Duplicate the `envvar` definition on line 366 and 368?

File: internal/pkg/admission/policy-server-deployment.go (hunk 4 above)
Comment (flavio) [reply internal/pkg/admission/policy-server-deployment.go]: You're right, `contain` is correct. However the name currently is "envVarsContain**t**Variable", note the highlighted "t". It should be `envVarsContainVariable`

File: internal/pkg/admission/policy-server-deployment.go (hunk 5 above)
Comment (flavio) [reply internal/pkg/admission/policy-server-deployment.go:366]: Scratch it... I don't know what I saw :facepalm:

File: internal/pkg/admission/policy-server-deployment.go (hunk 6 above)
Comment (flavio) [reply internal/pkg/admission/policy-server-deployment.go:372]: Same here... sorry about the noise

**Review by flavio — APPROVED** [review]
LGTM, there's just the typo in the function name to be fixed
